Assets/Character/Scripts/PlayerStateController.cs

```diff
@@ -2,59 +2,62 @@
 
 public class PlayerStateController : MonoBehaviour
 {
-	public Animator animator;
-
+	private Animator _animator;
 	private SpriteRenderer _spriteRenderer;
 	private bool _facingRight;
-	float _distToGround;
+
+	public int GetState()
+	{
+		return _animator.GetInteger("state");
+	}
 
 	void Start ()
 	{
-		animator = this.GetComponent<Animator>();
-		animator.SetInteger("state", Constants.STATE_IDLE);
+		_animator = GetComponent<Animator>();
+		_spriteRenderer = GetComponent<SpriteRenderer>();
+		_animator.SetInteger("state", Constants.STATE_IDLE);
 		_facingRight = true;
-		_spriteRenderer = this.GetComponent<SpriteRenderer>();
-		_distToGround = this.GetComponent<Collider2D>().bounds.extents.y;
 	}
 
 	void Update()
 	{
-		if (InState(Constants.STATE_IDLE))
-		{
-			IdleStateUpdate();
-		}
-		else if (InState(Constants.STATE_MID_ATTACK))
-		{
-			MidAttackStateUpdate();
-		}
-		else if (InState(Constants.STATE_MOVE_LEFT) || InState(Constants.STATE_MOVE_RIGHT))
-		{
-			MoveStateUpdate();
-		}
-		else if (InState(Constants.STATE_ROLL))
-		{
-			RollStateUpdate();
-		}
-		else if (InState(Constants.STATE_JUMP))
-		{
-			JumpStateUpdate();
-		}
-		else if (FinishedCurrentAnimation())
-		{
-			animator.SetInteger("state", Constants.STATE_IDLE);
+		switch (GetState())

```

Comment:
Consider creating a class for each player state and have a Map<Constants.STATE_ENUM, StateClass>. then call Map[Get(State)].update();